pkg/kubelet/dockertools/docker.go

```diff
@@ -153,6 +154,14 @@ func filterHTTPError(err error, image string) error {
 
 // Check if the inspected image matches what we are looking for
 func matchImageTagOrSHA(inspected dockertypes.ImageInspect, image string) bool {
+	// NB: the code below this check checks for validity of `image` as a pull spec.
+	// However, it can be useful to also inspect images by ID (config digest, e.g.
```

Comment:
So, the only way to get into the docker `InspectImage` API is through the `InspectImage` method on `DockerInterface`, which calls this.  So the alternative is to add an `InspectImageUnchecked` to `DockerInterface` which doesn't verify the inspected info after pulling (or add a `check` argument to the `InspectImage` method).

I see what you mean here, though.  The intent of the method was not entirely clear based on the godoc above :-/.